services/data/src/links/RestAPILink/fetchData.ts

```diff
@@ -15,26 +15,36 @@ export const parseStatus = async (response: Response) => {
         response.status === 403 ||
         response.status === 409
     ) {
-        const message = await response
-            .json()
-            .then(body => {
-                return body.message
-            })
-            .catch(() => {
-                return response.status === 401 ? 'Unauthorized' : 'Forbidden'
-            })
+        let message = ''
+        let details
```

Comment:
I wonder... should `details` be an empty object if the JSON is bad?  That way `error.details` will always be an object, never undefined.

Reply:
So I've pushed this 5ff6393, which fails on CI at the moment. I've set the type for details to an object, but since it's reassigned and then expected to contain a message the typescript compiler complains. What would be the best way to address that?

Reply:
See [the type definition](https://github.com/dhis2/app-runtime/pull/510/files#diff-b90ef08b9e29525a5ae5c77e44bd370dR2) and one of the fallbacks [here](https://github.com/dhis2/app-runtime/pull/510/files#diff-2f6d8eb1e7f7ef53661593022e859b46R19)

Reply:
Ok, pushed some changes that I think address this properly, but let me know if there are better ways to handle the typing here.